design/one-pager-stacks-security-isolation.md

```diff
@@ -241,13 +235,134 @@ permissionScope: Cluster
 
 ### Stack Install
 
-Since cluster stacks and namespace stacks have differing scopes of execution in the control plane, it will be wise to allow differing access to request their installation.
-A simple way to approach this would be to create different installation request CRDs for each and then use RBAC to restrict access to them by different roles.
+Since cluster stacks and namespace stacks have differing scopes of execution in the control plane, it will be wise to
+ allow differing access to request their installation. A simple way to approach this would be to create different
+ installation request CRDs for each and then use RBAC to restrict access to them by different roles.
+
 Currently, we use a single `StackRequest` type, but going forward we should differentiate by stack type:
 
-* `ClusterStackInstall`: requests the installation of a cluster stack
-* `StackInstall`: requests the installation of a namespace stack
+* `ClusterStackInstall`: requests the installation of a cluster stack who's resources are all cluster scoped
+* `StackInstall`: requests the installation of a namespace stack who's resources are all namespace scoped
+
+With these two separate types, an administrator can allow access to installing cluster stacks to only certain restricted
+ roles, then perhaps be more permissive with access to installing namespaced stacks.
+
+If the install scope type does not match what the stack author has declared within their stack metadata, an error status
+ should be set on the install resource and installation of the stack should not proceed.
+
+### RBAC
+
+We will rely on RBAC to further define a clear separation of concerns for infrastructure resources, and workspace
+ resources in a namespace. In our model we assume that any Cluster Stack resources as they are global will be for the
+ infrastructure persona, whereas any namespace stack resources will be for the namespace that that stack is installed
+ into, thus the application persona.
+
+At install time:
+ - Cluster Stack: generate cluster roles for infra persona that auto-aggregate to built-in roles for
+ infrastructure owners.
+ - Namespaced Stack: generate cluster roles for the workspace persona different permission
```

Comment:
```suggestion
 - Namespaced Stack: generate `Role` definitions for workspace admins, editor, and viewer roles
```
Unless you're suggesting that everyone becomes cluster-scoped, including namespace stacks? Looks like that is the case, and would definitely let us use the aggregated ClusterRoles, since aggregated Roles don't seem to be supported AFAIK.

Reply:
All roles become cluster scoped, but since we get aggregation(main reason), and permissions for a workspace are a function of admin and stack installer this seems ok. Workspace clusterroles would then be bound to their respective namespace with a roleBinding. I could kind of go either way on this. I’m tipped slightly to use kubernetes tooling for aggregation of clusterroles for its simplicity, but happy to be overruled if you prefer roles inside the workspace.

I think all admin level roles should roll up to crossplane-admin for example, regardless of if they’re namespace or cluster scoped.